extmod/modure.c

```diff
@@ -53,6 +53,10 @@ typedef struct _mp_obj_match_t {
     const char *caps[0];
 } mp_obj_match_t;
 
+STATIC mp_obj_t mod_re_compile(size_t n_args, const mp_obj_t *args);
+#if !MICROPY_ENABLE_DYNRUNTIME
+extern const mp_obj_type_t re_type;
```

Comment:
This can be STATIC, not extern

Reply:
Tried that already (in combination with the above STATIC remaining), but then data segment size goes up 32 bytes, as if there are 2 instances.

Reply:
I checked this.  On all ports it makes 0 size difference, except unix x64 which is 32 bytes larger in the data section.  This is due to alignment: putting the declaration here makes `re_type` appear in a different order in the data section and alignment means it grows by 32 bytes (4 words).

It's not worth worrying about these 32 bytes because it's a compiler-specific thing that we cannot control (that 32 bytes will one day be absorbed by something else somewhere).

So, for consistency with the rest of this file, I'd suggest to keep the STATIC.